connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2582,9 +2582,14 @@ class SparkConnectPlanner(
               .newBuilder()
               .setData(ByteString.copyFrom(bytes))))
     } else {
-      // Trigger assertExecutedPlanPrepared to ensure post ReadyForExecution before finished
-      // executedPlan is currently called by createMetricsResponse below
-      df.queryExecution.assertExecutedPlanPrepared()
+      if (df.isStreaming) {
```

Comment:
@jdesjean please review this PR.

Reply:
The approach used here is correct for the events.
However, if assertExecutedPlanPrepared should be avoided, should we remove it?

Reply:
Remove it. Thanks!